### PR TITLE
Np 3544 remove duplicate migration service

### PR DESCRIPTION
**Pull request by axthosarouris**
Both me and Jan implemented a Migration service to increase (work) parallelism. I delete now my interface to use Jan's

### CODE REVIEW

File: user-access-event-handlers/build.gradle [hunk 1]
```diff
@@ -6,6 +6,10 @@ plugins {
 dependencies {
     implementation project(":user-access-internal-model")
     implementation project(":user-access-commons")
+    implementation project(":user-access-service")
```
Comment (axthosarouris) [inline user-access-event-handlers/build.gradle:9]: organized dependencies and delete duplicate entries

File: user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/EventBasedScanHandler.java [hunk 2]
```diff
@@ -36,7 +38,7 @@ public EventBasedScanHandler() {
 
     public EventBasedScanHandler(AmazonDynamoDB dynamoDbClient,
                                  EventBridgeClient eventsClient,
-                                 MigrationService migrationService) {
+                                 UserMigrationService migrationService) {
```
Comment (axthosarouris) [inline user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/EventBasedScanHandler.java:41]: I had one interface and Jan had another, so I delete mine.

File: user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/EchoMigrationService.java [hunk 3]
```diff
@@ -1,10 +1,10 @@
-package no.unit.nva.useraccess.events;
+package no.unit.nva.useraccess.events.service;
 
 import java.util.ArrayList;
 import java.util.List;
 import no.unit.nva.useraccessmanagement.model.UserDto;
 
-public class EchoMigrationService implements MigrationService {
+public class EchoMigrationService implements UserMigrationService {
```
Comment (axthosarouris) [inline user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/EchoMigrationService.java:7]: This is a simple service for (online human) testing purposes to be sure that the scanner works as expected. This class will be deleted very soon. There is no dependency on this class in the tests.

File: user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/UserMigrationService.java [hunk 4]
```diff
@@ -1,9 +1,8 @@
 package no.unit.nva.useraccess.events.service;
 
-import no.unit.nva.useraccessmanagement.dao.UserDb;
+import no.unit.nva.useraccessmanagement.model.UserDto;
 
 public interface UserMigrationService {
 
-    UserDb migrateUser(UserDb user);
-
+    UserDto migrateUser(UserDto user);
```
Comment (axthosarouris) [inline user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/UserMigrationService.java:7]: migrationService should not be aware of the database objects (UserDb). It should instead operate on the public model (DTOs)

File: user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/UserMigrationServiceImpl.java [hunk 5]
```diff
@@ -18,7 +16,7 @@ public UserMigrationServiceImpl(CustomerService customerService) {
     }
 
     @Override
-    public UserDb migrateUser(UserDb user) {
+    public UserDto migrateUser(UserDto user) {
```
Comment (axthosarouris) [inline user-access-event-handlers/src/main/java/no/unit/nva/useraccess/events/service/UserMigrationServiceImpl.java:19]: see comment about DAOs and DTOS

File: user-access-event-handlers/src/test/java/no/unit/nva/useraccess/events/EventBasedScanHandlerTest.java [hunk 6]
```diff
@@ -203,7 +205,7 @@ private ScanDatabaseRequest createSampleScanRequest(Integer pageSize) {
         return new ScanDatabaseRequest(IDENTITY_SERVICE_BATCH_SCAN_EVENT_TOPIC, pageSize, NO_START_MARKER);
     }
 
-    private static class FamilyNameChangeMigration implements MigrationService {
+    private static class FamilyNameChangeMigration implements UserMigrationService {
```
Comment (axthosarouris) [inline user-access-event-handlers/src/test/java/no/unit/nva/useraccess/events/EventBasedScanHandlerTest.java:208]: This is a test class inside the Handler's unit tests. It performs a stupid migration to the users in order to verify that the users have indeed been migrated.

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

**Pull request merged**